### PR TITLE
[DOCS] Updated Databricks setup documentation

### PR DESCRIPTION
**Pull request by edurdevic**
## Is this PR related to a proposed Issue?

No

## What changes were proposed in this PR?

I updated the documentation about setting up Sedona in Databricks.

## How was this patch tested?

Yes, manual test on Databricks DBR 9.1 LTS

## Did this PR include necessary documentation updates?

Yes

### CODE REVIEW

**Comment by jiayuasu** [comment]
@edurdevic Thanks for updating this. Did you test Sedona SQL in both community edition and non-free edition? and Databricks runtime (7.x, 8.x, and 9.x)?

This was my observation:

For Databricks runtime 7.x,

Sedona RDD and SQL work in both community edition and advanced (non-free) edition. But the jar has to be manually uploaded to dbfs in advaned edition.

For Databricks runtime 8.x and 9.x, which use Spark 3.1+,

Sedona RDD and SQL work in community edition. But Sedona SQL join query will fail. The reason was unknown because Sedona was tested in official Spark 3.1+. So for the join query part, the user has to use RDD join query. I didn't test the advanced version (non-free version).

File: docs/download/databricks.md [hunk 1]
```diff
@@ -4,13 +4,66 @@ You just need to install the Sedona jars and Sedona Python on Databricks using D
...
+
+1) Insert a python (or scala) cell at the beginning of your SQL notebook to activate the bindings
+
+    ```Python
+    %python
+    from sedona.register.geo_registrator import SedonaRegistrator
+    SedonaRegistrator.registerAll(spark)
+    ```
+
+2) Install the sedona libraries from the [cluster init-scripts](https://docs.databricks.com/clusters/init-scripts.html) and activate the bindings by adding `spark.sql.extensions org.apache.sedona.viz.sql.SedonaVizExtensions,org.apache.sedona.sql.SedonaSqlExtensions` to your cluster's spark configuration. This way you can activate the Sedona bindings without typing any python or scala code.
```
[the remark this answers is not in the document]
Comment (edurdevic) [reply docs/download/databricks.md]: Yes, good point @alexott. 
I added some example scripts to be run in a notebook that will create the init script and download the dependencies via CURL. Please let me know if there is a better way to get the maven URL.

File: docs/download/databricks.md [hunk 2]
```diff
@@ -4,13 +4,66 @@ You just need to install the Sedona jars and Sedona Python on Databricks using D
...
+
+(or python)
+```Python
+from sedona.register.geo_registrator import SedonaRegistrator
+SedonaRegistrator.registerAll(spark)
+```
 
 ## Pure SQL environment
+ 
+In order to use the Sedona `ST_*` functions from SQL, you need to register the Sedona bindings. There are two ways to do that:
```
[the remark this answers is not in the document]
Comment (edurdevic) [reply docs/download/databricks.md]: Done, thanks!

**Comment by edurdevic** [comment]
> Thank you all for your help. We are working on the support of Spark 3.2 but were stuck at a minor issue. Let's see how it goes.
> 
> Meanwhile, as a temporary solution, users can mix their DataFrame API with RDD API to do a spatial join query. See the example here: https://github.com/apache/incubator-sedona/blob/master/binder/ApacheSedonaSQL_SpatialJoin_AirportsPerCountry.ipynb

Thanks @jiayuasu, I added this information in the docs.

**Comment by edurdevic** [comment]
@jiayuasu This PR looks accurate to me now. Do you think we can merge it?

**Comment by jiayuasu** [comment]
@edurdevic Thanks for reminding me. I was focusing on Spark 3.2 support, which is relevant to this doc. Will merge it now. Thank you!

**Pull request merged**